.github/workflows/test_develop.yml

```diff
@@ -16,12 +16,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
I don't see a reason why we should require numpy 2.0 in the first place. We only use very basic numpy features, that should be compatible with pretty much any version.

Reply:
TL/DR: Don't put a version constraint on numpy.

The dependencies (in the original `requirements.txt`) only specify `numpy` without a specific version constraint, which I think is the right thing to do here. (We could add a lower bound in case we use a feature that is not present in an earlier numpy version.) What your changes do is to pin a specific numpy version, which is not a good idea for a library. The user should decide, which numpy version to use, not the library.

If you want to pin your dependency versions for local development, you can use a different file (e.g. `requirements-local.txt`), but that doesn't need to be committed to git (unless we want to have some kind of standardized environment for development). The same goes for testing environments: If you need different dependencies for the versions, you could make a `requirements-<python-version>.txt` variant for each version, but I don't think that's necessary here if you keep `numpy` without version constraints. This will normally select the latest compatible version in each of the testing environments.

We could also consider specifying our library dependencies differently. Using `requirements.txt` is a kind of "quick and dirty" way to do that, but it's easily conflated with a local development environment (like here). We could either specify them in `setup.py` directly or switch to something more modern like `pyproject.toml`.

Reply:
After some more trying it seems that there are some numpy functionalities being used in the library that are only available in versions later than 2.0, like numpy.strings for example (in module spelled_array). I am not very familiar with this part of the code, but this does create a problem

Reply:
To clarify: `requirements.txt` should still contain `numpy`, but without a version constraint.

Reply:
That can't be a <2.0 issue. numpy 2.0 was released this summer, this library hasn't changed for two years. I'll have a look.

Reply:
Yes, the issue was caused by a change on this branch when trying to get numpy 2.0 to work. It's resolved in #26, so merging that (into this branch) should make you good to go.